homeassistant/components/enigma2/config_flow.py

```diff
@@ -115,23 +115,45 @@ async def async_step_user(
         if user_input is None:
             return self.async_show_form(step_id=SOURCE_USER, data_schema=CONFIG_SCHEMA)
 
-        data = await self.validate_user_input(user_input)
+        await self.validate_user_input(user_input)
         if "base" in self.errors:
             return self.async_show_form(
                 step_id=SOURCE_USER, data_schema=CONFIG_SCHEMA, errors=self.errors
             )
         return self.async_create_entry(
-            data=data, title=data[CONF_HOST], options=self._options
+            data=self._data, title=self._data[CONF_HOST], options=self._options
         )
 
     async def async_step_import(self, user_input: dict[str, Any]) -> ConfigFlowResult:
-        """Validate import."""
+        """Handle the import step."""
         if CONF_PORT not in user_input:
             user_input[CONF_PORT] = DEFAULT_PORT
         if CONF_SSL not in user_input:
             user_input[CONF_SSL] = DEFAULT_SSL
         user_input[CONF_VERIFY_SSL] = DEFAULT_VERIFY_SSL
 
+        self._data = {
+            key: user_input[key] for key in user_input if key in self.DATA_KEYS
+        }
+        self._options = {
+            key: user_input[key] for key in user_input if key in self.OPTIONS_KEYS
+        }
+
+        await self.validate_user_input(self._data)
+
+        if "base" in self.errors:
+            async_create_issue(
+                self.hass,
+                DOMAIN,
+                f"deprecated_yaml_{DOMAIN}_import_issue",
+                breaks_in_ha_version="2024.11.0",
+                is_fixable=False,
+                issue_domain=DOMAIN,
+                severity=IssueSeverity.WARNING,
+                translation_key="deprecated_yaml_import_issue",
+            )
+            return self.async_abort(reason=self.errors["base"])
+
         async_create_issue(
```

Comment:
This should only be raised when already configured or when it's the first time

Reply:
I think it will only raise the first time, because afterwards the import should be caught by the abort criteria in the `validate_user_input` function.